tests/PregWrapperTest.php

```diff
@@ -0,0 +1,133 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Tests;
+
+use PhpCsFixer\PregWrapper;
+
+/**
+ * @author Kuba Werłos <werlos@gmail.com>
+ *
+ * @internal
+ *
+ * @covers \PhpCsFixer\PregWrapper
+ */
+final class PregWrapperTest extends TestCase
+{
+    /**
+     * @param string $method
+     * @param string $function
+     * @param array  $arguments
+     *
+     * @dataProvider provideFailingCases
+     */
+    public function testFailing($method, $function, array $arguments)
+    {
+        $this->expectException(\RuntimeException::class);
+        $this->expectExceptionMessage(sprintf('Error occurred when calling %s.', $function));
+        $this->expectExceptionCode(PREG_BACKTRACK_LIMIT_ERROR);
+
+        call_user_func_array(PregWrapper::class.'::'.$method, array_merge(['/(?:\D+|<\d+>)*[!?]/'], $arguments));
+    }
+
+    public function provideFailingCases()
+    {
+        return [
+            ['match', 'preg_match', ['foobar foobar foobar']],
```

Comment:
Usually, a test method only covers a single method of the tested class and I must admit I don't feel very comfortable with having a tested method per test case instead. But if others are ok with that, then 👍 for me.

Reply:
When I started working on the wrapper I wrote tests first and this way was more readable, in the cost of "following the art". `PregWrapper` is also in fact "ifology" and structural code (written as class), not OO code.